content/js/query.md

```diff
@@ -136,10 +117,83 @@ the relation with the current user.
 For example, to query for records owned by a user that the current user is following:
 
 ```javascript
-query.havingRelation("_owner", skygear.relation.Following)
+query.havingRelation("_owner", skygear.relation.Following);
+query.notHavingRelation(...);
+```
+
+See [Social](/js/guide/relation) section for more relations.
+
+### Geolocation condition
+
+Please visit [Geo-location](/js/guide/geolocation) section for more.
+
+<a name="pagination-ordering"></a>
+## Pagination and Ordering
+
+### Ordering of records
+
+We can sort the records based on certain field in ascending or descending order.
+
+``` javascript
+query.addAscending('age');    // sorted by age increasing order
+query.addDescending('price'); // sorted by price decreasing order
+```
+
+### Pagination of records
+
+We can limit the number of records returned, skip certain number of records,
+and access certain part of records via pagination.
+
+``` javascript
+query.limit = 10; // return 10 records only
+query.offset = 15; // skip the first 10 records
+/* The above query will return the 16th to 25th records */
+
+query.page = 3; // skip the first two pages of records
+// number of records per page equals limit
+// if limit is unset, default limit is 50
+/* The above query will return the 101th to 150th records */
+```
+
+### Counting the records
+
+To get the number of records matching a query, set the `overallCount`
+of the Query to `true`.
+
+``` javascript
+query.overallCount = true;
+skygear.privateDB.query(query).then((notes, count) => {
+  console.log('%d records matching query.', count);
+}, (error) => {
+  console.error(error);
+});
 ```
 
-See [User Relations (JS SDK)](/js/guide/relation) for more relations.
+The count is not affected by the limit set on the query. So, if you only want
+to get the count without fetching any records, simply set `query.limit = 0`.
+
+<a name="relational-query"></a>
+## Relational Queries
+
+### Eager Loading
+
+If you have a record that references another record, you can perform eager
+loading using the transient syntax. It is possible to eager load records from
+multiple keys, but doing so will impair performance. If you don't know about
+reference, please read [Records](/js/guide/record#reference) section first.
+
+``` javascript
+let query = new skygear.Query(Note);
+query.transientInclude('nextPage', '<your-transient-name>');
```

Comment:
- Here needs more explanation of the function signature `transientInclude(key, mapToKey)`.
- `nextPage` does not sound like an easy-to-understand relationship. How about another attribute?